dss/operations/secrets.py

```diff
@@ -291,3 +293,163 @@ def del_secret(argv: typing.List[str], args: argparse.Namespace):
+    """
+    def __init__(self, stage):
+        """Initialize variables useful to the secrets checker"""
+        self.stage = stage
+        self.stages = {'dev': 'environment',
+                       'integration': 'environment.integration',
+                       'staging': 'environment.staging'}
+
+        self.missing_secrets = []
+        self.malformed_secrets = []
+        self.incomplete_secrets = []
+        self.error_message = f'\n\n' \
+                             f'Deploying to {self.stage.upper()} could not be completed.\n' \
+                             f'It looks like an AWS secret has an unexpected value.\n' \
+                             f'Please do not change AWS secrets for releases.\n'
+
+        if self.stage not in self.stages:
+            print(f'Custom stage "{self.stage}" provided. Secret checking skipped.')
+            return
+
+        self.stage_env = copy.deepcopy(os.environ)
+        self.stage_env = self.get_stage_env(self.stages[self.stage])
+        self.service_account = self.fetch_terraform_output("service_account", "gcp_service_account").strip()
+
+        self.email = [f'{self.service_account}@human-cell-atlas-travis-test.iam.gserviceaccount.com']
+        self.project = ['human-cell-atlas-travis-test']
+        self.type = ['service_account']
+        self.auth_uri = ['https://auth.data.humancellatlas.org/oauth/authorize',
+                         'https://auth.dev.data.humancellatlas.org/oauth/authorize']
+        self.token_uri = ['https://auth.data.humancellatlas.org/oauth/token',
+                          'https://auth.dev.data.humancellatlas.org/oauth/token']
+
+        self.app_secret_name = os.environ['GOOGLE_APPLICATION_SECRETS_SECRETS_NAME']
+        self.gcp_cred_secret_name = os.environ['GOOGLE_APPLICATION_CREDENTIALS_SECRETS_NAME']
+        self.app_secret = self.fetch_secret(self.app_secret_name)
+        self.gcp_cred_secret = self.fetch_secret(self.gcp_cred_secret_name)
+
+    def run_cmd(self, cmd, cwd=os.getcwd(), shell=True):
+        """Run a command and return stdout"""
+        p = subprocess.Popen(cmd,
+                             shell=shell,
+                             stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE,
+                             cwd=cwd,
+                             env=self.stage_env)
+        stdout, stderr = p.communicate()
+        if p.returncode != 0:
+            raise RuntimeError(f'While checking secrets, an error occured:\n'
+                               f'stdout: {stdout}\n\n'
+                               f'stderr: {stderr}\n')
+        return stdout.decode('utf-8')
+
+    def get_stage_env(self, env_file):
+        """Source the environment file for this stage, then export the current environment as a dict"""
+        dump = 'python -c "import os, json; print(json.dumps(dict(os.environ)))"'
+        cmd = ['bash', '-c', f'source {env_file} && {dump}']
```

Comment:
For sourcing `dev`
`source environment`

For sourcing `integration`
`source environment && source environment.integration`

Reply:
Does the CI pipeline that runs the stage already have the first environment file sourced? if so i think all we would need is a comment about this.....

Reply:
The `environment` and `environment.$STAGE` files should be sourced before this test is run (see https://github.com/HumanCellAtlas/data-store/blob/master/.gitlab-ci.yml#L26-L29), so this is not an issue when running this as a test. 

However, we should make sure that if `scripts/dss-ops.py secrets check` is run independent of tests, it still sources environment files in the correct order...

Reply:
As it is written currently, it will not source environment files in the correct order - non-dev stages will only source `environment.$STAGE` and will not source `environment` first

Reply:
Fixed by changing the command from `source {env_file} && {dump}` to `source environment && source {env_file} && {dump}`

Reply:
Fixed by changing the command from `source {env_file} && {dump}` to `source environment && source {env_file} && {dump}`